iris_grib/tests/unit/load_convert/test_product_definition_section.py

```diff
@@ -0,0 +1,134 @@
+# (C) British Crown Copyright 2014 - 2019, Met Office
+#
+# This file is part of iris-grib.
+#
+# iris-grib is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# iris-grib is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with iris-grib.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Tests for `iris_grib._load_convert.product_definition_section`.
+
+"""
+
+from __future__ import (absolute_import, division, print_function)
+
+# import iris_grib.tests first so that some things can be initialised
+# before importing anything else.
+import iris_grib.tests as tests
+
+import mock
+from iris.coords import DimCoord
+from iris_grib._load_convert import product_definition_section
+
+from iris_grib.tests.unit.load_convert import empty_metadata
+from iris_grib.tests.unit.load_convert.test_product_definition_template_0 \
+    import section_4 as template_0
+from iris_grib.tests.unit.load_convert.test_product_definition_template_31 \
+    import section_4 as template_31
+
+
+class Test(tests.IrisGribTest):
+    def setUp(self):
+        self.patch('warnings.warn')
+        self.translate_phenomenon_patch = self.patch(
+            'iris_grib._load_convert.translate_phenomenon'
+        )
+
+
+# Tests focussing on the handling of fixed surface elements in section 4.
+# Expects/ignores depending on the template number.
+class TestFixedSurfaces(Test):
+    def generic_fixed_surface_test(self, fs_is_expected, fs_is_present):
+        # Whether or not fixed surface elements are expected/present in the
+        # section 4 keys, most of the code is shared so we are using a single
+        # function with parameters.
+
+        # Prep placeholder variables for product_definition_section.
+        discipline = mock.sentinel.discipline
+        tablesVersion = mock.sentinel.tablesVersion
+        rt_coord = DimCoord(24, 'forecast_reference_time',
+                            units='hours since epoch')
+        metadata = empty_metadata()
+
+        # Use the section 4 from either product_definition_section #1 or #31.
+        # #1 contains fixed surface elements, #31 does not.
```

Comment:
@trexfeathers This should be `#0` and not `#1` me thinks :wink:

Reply:
Excellent eyes! 🦅